pkg/auth/authz_handler.go

```diff
@@ -118,6 +120,24 @@ func OrgIDFromContext(ctx context.Context) string {
 	return payload.Organization
 }
 
+// EmailFromContext returns email from the specified context
+func EmailFromContext(ctx context.Context) string {
+	payload := PayloadFromContext(ctx)
+	return payload.Email
+}
+
+// EmailDomainFromContext returns email Domain from the specified context
+func EmailDomainFromContext(ctx context.Context) string {
+	domain := "Unknown"
+	email := EmailFromContext(ctx)
+	delimiterIdx := strings.LastIndex(email, EmailDelimiter)
+	if delimiterIdx >= 0 {
+		emailElements := strings.Split(email, "@")
```

Comment:
why not to split and then check the size of the array?

Reply:
what cases will it catch that this method won't?

Reply:
just searching for the string twice is less efficiant...
Nothing too important

Reply:
multiple `@` signs are allowed in an email address, in which the last `@` should be used as the delimiter between the local part and domain. This is why I used `LastIndex`.

quote:
`
space and special characters "(),:;<>@[\] are allowed with restrictions (they are only allowed inside a quoted string, as described in the paragraph below, and in addition, a backslash or double-quote must be preceded by a backslash);
`

source: https://en.wikipedia.org/wiki/Email_address#Local-part